### PR TITLE
Feature/skip deleted

### PR DESCRIPTION
**Pull request by no-reply**
- Add harvest behavior objects to harvester. This makes it possible to introduce new record processing behavior by injecting them into the harvesters.
- Create skip deleted behavior for OAI, make it the default behavior for OAI harvesters.

### CODE REVIEW

**Comment by markbreedlove** [comment]
@no-reply and I have discussed this, and I think it looks good to merge.

I am about to make some compatible changes related to indexing and mapping activities, and will have a PR ready soon.

We might change where the behavior classes are located, but we're going to defer doing so until both of our branches are merged.

File: lib/krikri/harvester.rb [hunk 1]
```diff
@@ -165,6 +177,18 @@ def self.expected_opts
       raise NotImplementedError
     end
 
+    protected
```
Comment (anarchivist) [inline lib/krikri/harvester.rb]: Do we still need to retain this commented-out protected method?

**Comment by anarchivist** [comment]
I'm :+1: on this; just curious if we're planning to retain the commented method. Fix up and :squash: as appropriate.

**Comment by no-reply** [comment]
I can fix up and squash once I'm in the air.

**Comment by anarchivist** [comment]
:airplane: :jack_o_lantern:

**Comment by no-reply** [comment]
I think this is ready. :rocket:

**Comment by anarchivist** [comment]
:100:

**Pull request merged**